synapse/lib/storm.py

```diff
@@ -1104,15 +1104,14 @@ def _stormOperJoin(self, query, oper):
 
         args = oper[1].get('args')
         opts = dict(oper[1].get('kwlist'))
+        if len(args) != 2:
+            raise s_common.BadSyntaxError(mesg='join(<dstprop>,<srcprop>)')
 
         dstp = args[0]
-        srcp = args[0]
-
-        if len(args) > 1:
-            srcp = args[1]
+        srcp = args[1]
 
         # use the more optimal "in" mechanism once we have the pivot vals
-        vals = list({t[1].get(srcp) for t in query.data() if t is not None})
+        vals = list({t[1].get(srcp) for t in query.data() if t is not None and t[1].get(srcp) is not None})
```

Comment:
Prevents `None` from getting into the `vals` if the `srcp` is missing from `query.data()` (this usually results in the overall query failing because of an uncaught exception in a call to norm)

Reply:
Good call.  We should keep this either way.